interactive/views.py

```diff
@@ -191,7 +194,66 @@ def get_object(self, queryset=None):
 
     def get_context_data(self, **kwargs):
         report = process_html(self.object.report_content)
+        update_url = reverse(
+            "interactive:report-update",
+            kwargs=dict(
+                slug=self.object.slug,
+                project_slug=self.object.project.slug,
+                org_slug=self.object.project.org.slug,
+            ),
+        )
 
         return super().get_context_data(**kwargs) | {
             "report": report,
+            "update_url": update_url,
         }
+
+
+class AnalysisReportUpdate(View):
+    """Updating the title and description of the generated report.
+
+    It actually updates the attached Report model, rather than the
+    AnalysisRequest itself.
+    """
+
+    form_class = modelform_factory(Report, fields=["title", "description"])
+
+    def dispatch(self, request, slug, *args, **kwargs):
+        analysis_request = get_object_or_404(AnalysisRequest, slug=slug)
+
+        if not has_permission(
+            self.request.user, "analysis_request_view", project=analysis_request.project
+        ):
+            raise PermissionDenied
+
+        if not analysis_request.report:
+            raise Http404
+
+        self.analysis_request = analysis_request
+
+        return super().dispatch(request, slug, *args, **kwargs)
+
+    def get(self, request, *args, **kwargs):
+        form = self.form_class(instance=self.analysis_request.report)
+        return self.render_to_response(request, form)
+
+    def post(self, request, *args, **kwargs):
+        form = self.form_class(request.POST, instance=self.analysis_request.report)
+        if form.is_valid():
+            form.save()
+            project = self.analysis_request.project
+            return redirect(
+                "interactive:analysis-detail",
+                org_slug=project.org.slug,
+                project_slug=project.slug,
+                slug=self.analysis_request.slug,
+            )
+        else:
+            return self.render_to_response(request, form)
+
+    def render_to_response(self, request, form):
+        return TemplateResponse(
+            request,
+            template="interactive/report_update.html",
+            context={"form": form, "analysis_request": self.analysis_request},
+        )
```

Comment:
Rather than building the view around an AnalysisRequest we can focus it on the Report and lean on UpdateView:

```
class ReportEdit(UpdateView):
    fields = [
        "title",
        "description",
    ]
    model = Report
    template_name = "interactive/report_edit.html"

    def get_object(self, **kwargs):
        # keep hold of the AnalysisRequest object so we can easily use it for
        # the success_url
        self.analysis_request = get_object_or_404(
            AnalysisRequest, slug=self.kwargs["slug"]
        )

        if not has_permission(
            self.request.user,
            "analysis_request_view",
            project=self.analysis_request.project,
        ):
            raise PermissionDenied

        if not self.analysis_request.report:
            raise Http404

        return self.analysis_request.report

    def get_success_url(self):
        return self.analysis_request.get_absolute_url()
```

I'm returning the report, after the relevant checks, so the form is handled by the view for us.  I'm also keeping a reference on the AnalysisRequest so we don't have to put checks around `report.analysis_request` in get_success_url.

Could we also rename it and the template to use the edit suffix?  It's what we've done for all the other views (except the review request which I must have missed in the spike!).

Reply:
I tried to make this work originally, but it seemed a bit messy. But your way looks cleaner than what I got to.

I also had a concern here about normal Reports vs OSI Reports. Are we going to allow editing title/description on normal reports?

We can maybe punt on that decision. I guess its fine to have multiple urls that update Report objects.

Reply:
I think punt on the OSI vs normal report bits until we have some time to think about how reports will exist in job-server 👍